assets/js/modules/analytics/components/settings/GA4SettingsControls.js

```diff
@@ -142,14 +227,12 @@ export default function GA4SettingsControls( props ) {
 					) }
 			</div>
 
-			{ isDisabled && (
+			{ isDisabled ? (
 				<GA4ActivateSwitch
 					disabled={ ! hasAnalyticsAccess }
 					onActivate={ onActivate }
 				/>
-			) }
-
-			{ ! isDisabled && (
+			) : (
```

Comment:
Unrelated refactor that was needed to satisfy the cyclometic complexity eslint rules. :sweat_smile:

Reply:
Just chipping in here to say, I think disabling the complexity rule here would be the wrong approach - it would be necessary to disable it for the whole function, meaning this function could continue to grow more complex with no limit to it. It seems a bit of a slippery slope to start disabling the rule like this.

If this refactor is objectionable, I'd suggest taking a different approach to refactoring the function rather than disabling the rule entirely...

Reply:
@techanvil @tofumatt yep, disabling the rule is per function and not per line so disabling the rule will really bite us in the future. I can try to find other way to claim back another execution path without this specific refactor, but as far as I remember this was the most straightforward and reasonable one :sweat_smile:

Reply:
I actually don't mind the ternary here, but @tofumatt does seem quite strongly against it and I don't have a strong enough opinion on the matter to argue the point :)

Reply:
Well, I had to extract out the error related logic to a separate component anyway, as there were a few more changes that came from other merged PRs that put the complexity to the roof. Hence, I have reverted this change here but made @tofumatt's life harder by introducing another component and corresponding tests :sweat_smile: